crates/oxc_semantic/src/scoping.rs

```diff
@@ -677,6 +677,12 @@ impl Scoping {
         });
     }
 
+    pub fn remove_child_scope(&mut self, scope_id: ScopeId, child_scope_ids: ScopeId) {
+        self.cell.with_dependent_mut(|_allocator, cell| {
+            cell.scope_child_ids[scope_id.index()].retain(|scope_id| *scope_id != child_scope_ids);
+        });
```

Comment:
What I was trying to say in previous comment is that this can be more efficient than `remove_child_scopes`, because you're only removing a single ID.

```rs
let child_ids = &mut cell.scope_child_ids[scope_id.index()];
let index = child_ids.iter().position(|&scope_id| scope_id == child_scope_id).unwrap();
child_ids.swap_remove(index);
```

Note that `retain` shifts up all following entries in the `Vec` (slow if there's lots of them), whereas `swap_remove` just moves 1 entry (same cheap cost, no matter how many entries in the `Vec`).

Warning: The `unwrap()` will panic if the `ScopeId` isn't found. That means this method must not be called unless the option to record child scope IDs is enabled. Method needs a comment to that effect.

Reply:
ahhh i see, thanks for explaining